include/mockturtle/views/cost_view.hpp

```diff
@@ -0,0 +1,350 @@
+class cost_view<Ntk, RecCostFn, context_t, false> : public Ntk
+{
+public:
+  using storage = typename Ntk::storage;
+  using node = typename Ntk::node;
+  using signal = typename Ntk::signal;
+  using costfn_t = RecCostFn;
+
+  explicit cost_view( RecCostFn const& cost_fn = {}, cost_view_params const& ps = {} )
+      : Ntk(), _ps( ps ), context( *this ), _cost_fn( cost_fn )
+  {
+    static_assert( is_network_type_v<Ntk>, "Ntk is not a network type" );
+    static_assert( has_size_v<Ntk>, "Ntk does not implement the size method" );
+    static_assert( has_is_complemented_v<Ntk>, "Ntk does not implement the is_complemented method" );
+    static_assert( has_visited_v<Ntk>, "Ntk does not implement the visited method" );
+    static_assert( has_set_visited_v<Ntk>, "Ntk does not implement the set_visited method" );
+    static_assert( has_foreach_pi_v<Ntk>, "Ntk does not implement the foreach_pi method" );
+
+    add_event = Ntk::events().register_add_event( [this]( auto const& n ) { on_add( n ); } );
+  }
+
+  /*! \brief Standard constructor.
+   *
+   * \param ntk Base network
+   */
+  explicit cost_view( Ntk const& ntk, RecCostFn const& cost_fn = {}, cost_view_params const& ps = {} )
+      : Ntk( ntk ), _ps( ps ), context( ntk ), _cost_fn( cost_fn )
+  {
+    static_assert( is_network_type_v<Ntk>, "Ntk is not a network type" );
+    static_assert( has_size_v<Ntk>, "Ntk does not implement the size method" );
+    static_assert( has_is_complemented_v<Ntk>, "Ntk does not implement the is_complemented method" );
+    static_assert( has_visited_v<Ntk>, "Ntk does not implement the visited method" );
+    static_assert( has_set_visited_v<Ntk>, "Ntk does not implement the set_visited method" );
+    static_assert( has_foreach_pi_v<Ntk>, "Ntk does not implement the foreach_pi method" );
+
+    update_cost();
+    add_event = Ntk::events().register_add_event( [this]( auto const& n ) { on_add( n ); } );
+  }
+
+  /*! \brief Copy constructor. */
+  explicit cost_view( cost_view<Ntk, RecCostFn, context_t, false> const& other )
+      : Ntk( other ), _ps( other._ps ), context( other.context ), _cost_fn( other._cost_fn )
+  {
+    add_event = Ntk::events().register_add_event( [this]( auto const& n ) { on_add( n ); } );
+  }
+
+  cost_view<Ntk, RecCostFn, context_t, false>& operator=( cost_view<Ntk, RecCostFn, context_t, false> const& other )
+  {
+    /* delete the event of this network */
+    Ntk::events().release_add_event( add_event );
+
+    /* update the base class */
+    this->_storage = other._storage;
+    this->_events = other._events;
+
+    /* copy */
+    _ps = other._ps;
+    context = other.context;
+    _cost_fn = other._cost_fn;
+
+    add_event = Ntk::events().register_add_event( [this]( auto const& n ) { on_add( n ); } );
+
+    return *this;
+  }
+
+  ~cost_view()
+  {
+    Ntk::events().release_add_event( add_event );
+  }
+
+  /**
+   * @brief Get the context of a node
+   * 
+   * @param n 
+   * @return context_t context
+   */
```

Comment:
Why are these documentation comment in a different syntax? Trying to compile the documentation (`make html` in `docs/`), none of the member functions are actually compiled and shown on the generated webpage.